main_board/src/optics/ir_camera_system/unit_tests/ir_camera_timer_settings/main.c

```diff
@@ -442,53 +305,50 @@ ZTEST(timer_settings_on_time,
     zassert_equal(ts.on_time_in_us, settings.on_time_in_us,
                   "must be on_time_us");
     zassert_equal(fps, ts.fps, "must be %u, actual %u", fps, ts.fps);
-    zassert_equal(settings.psc, ts.psc,
-                  "must be unchanged, changed from %u to %u", settings.psc,
-                  ts.psc);
-    zassert_equal(settings.arr, ts.arr,
-                  "must be unchanged, changed from %u to %u", settings.arr,
-                  ts.arr);
-    zassert_equal(settings.ccr, ts.ccr,
-                  "must be 120%% of original, changed from %u to %u",
-                  settings.ccr, ts.ccr);
+    zassert_equal(settings.master_psc, ts.master_psc,
+                  "must be unchanged, changed from %u to %u",
+                  settings.master_psc, ts.master_psc);
+    zassert_equal(settings.master_arr, ts.master_arr,
+                  "must be unchanged, changed from %u to %u",
+                  settings.master_arr, ts.master_arr);
 }
 
-ZTEST(timer_settings_on_time, test_on_time_set_very_low_when_fps_is_at_minimum)
-{
-    struct ir_camera_timer_settings settings = {0};
-    struct ir_camera_timer_settings ts = {0};
-    uint32_t on_time_us, fps;
-    ret_code_t ret;
-
-    fps = 1;
-    on_time_us = 10;
-
-    ret = timer_settings_from_on_time_us(on_time_us, &settings, &ts);
-    zassert_equal(RET_SUCCESS, ret, "");
-    zassert_equal(ts.on_time_in_us, on_time_us, "must be on_time_us");
-    zassert_equal(0, ts.fps, "must be 0, actual %u", ts.fps);
-    zassert_equal(0, ts.psc, "must be 0, actual %u", ts.psc);
-    zassert_equal(0, ts.arr, "must be 0, actual %u", ts.arr);
-    zassert_equal(0, ts.ccr, "must be 0, actual %u", ts.ccr);
-
-    // the calculated ccr would be 0 but it should be capped to 1
-    ret = timer_settings_from_fps(fps, &ts, &ts);
-    zassert_equal(RET_SUCCESS, ret, "");
-    zassert_equal(ts.on_time_in_us, on_time_us, "must be on_time_us");
-    zassert_equal(ts.fps, fps, "must be %u, actual %u", fps, ts.fps);
-    zassert_not_equal(0, ts.psc, "must not be 0, actual %u", ts.psc);
-    zassert_not_equal(0, ts.arr, "must not be 0, actual %u", ts.arr);
-    zassert_equal(1, ts.ccr, "must be 1, actual %u", ts.ccr);
-
-    // same should apply for setting the on-time at 1 fps
-    ret = timer_settings_from_on_time_us(on_time_us, &ts, &ts);
-    zassert_equal(RET_SUCCESS, ret, "");
-    zassert_equal(ts.on_time_in_us, on_time_us, "must be on_time_us");
-    zassert_equal(ts.fps, fps, "must be %u, actual %u", fps, ts.fps);
-    zassert_not_equal(0, ts.psc, "must not be 0, actual %u", ts.psc);
-    zassert_not_equal(0, ts.arr, "must not be 0, actual %u", ts.arr);
-    zassert_equal(1, ts.ccr, "must be 1, actual %u", ts.ccr);
-}
+// ZTEST(timer_settings_on_time,
```

Comment:
should this test be removed?

Reply:
yes, it doesnt make sence with the new timer structure, as the ontime timers are independent from the fps timer

Reply:
I'll remove it